monai/transforms/signal/array.py

```diff
@@ -0,0 +1,773 @@
+# Copyright (c) MONAI Consortium
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+A collection of transforms for signal operations
+https://github.com/Project-MONAI/MONAI/wiki/MONAI_Design
+"""
+
+from typing import Any, List, Optional
+
+import numpy as np
+
+from monai.transforms.transform import RandomizableTransform, Transform
+from monai.utils import optional_import
+from monai.utils.enums import TransformBackends
+
+zoom, has_zoom = optional_import("scipy.ndimage", name="zoom")
+resample_poly, has_resample_poly = optional_import("scipy.signal", name="resample_poly")
+fft, has_resample_fft = optional_import("scipy.signal", name="resample")
+shift, has_shift = optional_import("scipy.ndimage.interpolation", name="shift")
+square, has_square = optional_import("scipy.signal", name="square")
+iirnotch, has_iirnotch = optional_import("scipy.signal", name="iirnotch")
+filtfilt, has_filtfilt = optional_import("scipy.signal", name="filtfilt")
+spectrogram, has_spectrogram = optional_import("scipy.signal", name="spectrogram")
+central_frequency, has_central_frequency = optional_import("pywt", name="central_frequency")
+cwt, has_cwt = optional_import("pywt", name="cwt")
+
+
+__all__ = [
+    "SignalRandDrop",
+    "SignalRandScale",
+    "SignalRandShift",
+    "SignalResample",
+    "SignalRandAddSine",
+    "SignalRandAddSquarePulse",
+    "SignalRandAddGaussianNoise",
+    "SignalRandAddSinePartial",
+    "SignalRandAddSquarePulsePartial",
+    "SignalNormalize",
+    "SignalStandardize",
+    "SignalZeroPad",
+    "SignalFillEmpty",
+    "SignalRemoveFrequency",
+    "SignalShortTimeFourier",
+    "SignalContinousWavelet",
+]
+
+
+class SignalResample(Transform):
+    """
+    Resample signal to the target sampling rate.
+    """
+
+    backend = [TransformBackends.NUMPY]
+
+    def __init__(
+        self, method: str = "interpolation", current_sample_rate: int = 500, target_sample_rate: int = 250
```

Comment:
500 and 250 seem to me to be completely arbitrary, is that correct? If so, I think the defaults should be removed.

Reply:
yes you're correct, i'll fix that

Reply:
fixed